llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -1597,7 +1597,6 @@ void NVPTXAsmPrinter::emitFunctionParamList(const Function *F, raw_ostream &O) {
             // CUDA kernels assume that pointers are in global address space
             // See:
             // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parameter-state-space
-            assert(addrSpace == 0 && "Invalid address space");
```

Comment:
I think this assertion is still (partially?) valid. The only case when we want to emit `.global` is if the pointer is generic or explicitly in the global AS.